### PR TITLE
Fix EffDrop overwriting values of merged xp orbs.

### PR DESCRIPTION
**Pull request by sovdeeth**
### Description

Maintains existing experience orb experience when setting the exp value in EffDrop. This fixes a bug with the Spigot exp orb merging that cause multiple orbs spawned together to overwrite each other's values, leaving, for example, 1 orb with 10xp instead of 1 orb with 50xp when spawning 5 orbs of 10xp each.

Edit:
Adds a ExprTotalExperience expression, as it wasn't possible to write a regression test without it.
Added util methods in PlayerUtils for getting and setting xp values. 
Also fixed the bug for `spawn 10 xp` as well. 

---
**Target Minecraft Versions:** 
**Requirements:** 
**Related Issues:** #5490

### CODE REVIEW

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]
Add a regression test to this

**Comment by sovdeeth** [comment]
> Add a regression test to this

How? We don't have syntax to get the amount of xp of an orb, right?

**Comment by TheLimeGlass** [comment]
https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/expressions/ExprExperience.java but it won't work until the JUnit is merged

**Comment by sovdeeth** [comment]
> https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/expressions/ExprExperience.java but it won't work until the JUnit is merged

Can you give me an example/link me to how this would work? Since that expression only works in events, i'm not sure how to make a test out of it.

**Comment by TheLimeGlass** [comment]
> > https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/expressions/ExprExperience.java but it won't work until the JUnit is merged
> 
> Can you give me an example/link me to how this would work? Since that expression only works in events, i'm not sure how to make a test out of it.

JUnit testing system now implemented

**Comment by sovdeeth** [comment]
Ended up writing a new total experience expression, as the spawn/drop effects didn't trigger experience spawn events. Plus, it lets me write a normal test instead of JUnit trickery.

File: src/main/java/ch/njol/skript/expressions/ExprTotalExperience.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+			return PlayerUtils.getTotalXP(((Player) entity).getLevel(), ((Player) entity).getExp());
+
+		// invalid entity type
+		return null;
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE || mode == ChangeMode.RESET || mode == ChangeMode.DELETE)
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/expressions/ExprTotalExperience.java:73]: Why don't you make this negated for REMOVE_ALL

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprTotalExperience.java (hunk 1 above)
Comment (Fusezion) [reply src/main/java/ch/njol/skript/expressions/ExprTotalExperience.java:73]: unless I miss understanding I believe this falls in line with this comment https://github.com/SkriptLang/Skript/pull/5513#discussion_r1133405621

Comment (TheLimeGlass) [reply src/main/java/ch/njol/skript/expressions/ExprTotalExperience.java:73]: Then a switch should be in place

**Review by TheLimeGlass — APPROVED** [review]

**Pull request merged**